app/scripts/controllers/sponsormodal.js

```diff
@@ -1,20 +1,63 @@
 'use strict';
 
 angular.module('bulbsCmsApp')
-  .controller('SponsormodalCtrl', function ($scope, ContentFactory, article, Campaign) {
+  .controller('SponsormodalCtrl', function ($scope, ContentFactory, article, Campaign,
+                                            // NOTE: temp depdendencies will be moved along with searchCampaigns/loadCampaigns
+                                            _, $q, $http, TunicConfig) {
     $scope.article = article;
 
-    if ($scope.article.campaign) {
-      $scope.campaign = Campaign.$find($scope.article.campaign);
+    $scope.tunicCampaignIdMapping = {};
+
+    // --------------------------- TODO: MOVE ME SOMEWHERE ELSE---------------------------------------
+    var $loadTunicCampaign = function (tunicCampaignId) {
```

Comment:
Where to put this (and $searchCampaigns)? Special Coverage implementation added this to model inside special-coverage-factory.js). Or better to share this logic somehow?